File: leetcode/1823/findTheWinnerOfTheCircularGame.swift
class Solution {
    func findTheWinner(_ n: Int, _ k: Int) -> Int {
        var start = 0
        var lst : [Int] = []
        var idx : Int
        for i in 1...n{
            lst.append(i)
        }

        while lst.count > 1 {
            idx = (start + k - 1) % lst.count
            lst.remove(at:idx)
            start = idx
        }
        return lst[0]
    }
}

File: leetcode/1823/findTheWinnerOfTheCircularGame.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        lst = [i for i in range(1, n+1)]
        start = 0

        while len(lst) > 1:
            idx = (start + k -1) % len(lst) 
            lst.remove(lst[idx])

            start = idx % n

        return lst[0]
